src/components/project_card.rs

```diff
@@ -0,0 +1,108 @@
+use yew::{html, Component, ComponentLink, Html, Properties, ShouldRender};
+
+#[derive(Clone, Copy,PartialEq, Eq, Hash)]
+pub enum Lang {
+    Rust,
+    CSS,
+    JavaScript,
+    HTML,
+    NoLanguage
+}
+
+impl Lang {
+    /// Retrieve the class and the text for the repository
+    /// language field as a tuple where `0` is the class name
+    /// and `1` the label name
+    pub fn markup_resources(&self) -> (String, String) {
+        match self {
+            Lang::Rust => (String::from("rust"), String::from("Rust")),
+            Lang::CSS => (String::from("css"), String::from("CSS")),
+            Lang::JavaScript => (String::from("javascript"), String::from("JavaScript")),
+            Lang::HTML => (String::from("html"), String::from("HTML")),
+            Lang::NoLanguage => (String::default(), String::from("No language")),
+        }
+    }
+}
+
+impl From<String> for Lang {
+    fn from(string: String) -> Self {
+        match string.to_lowercase().as_str() {
+            "rust" => Lang::Rust,
+            "html" => Lang::HTML,
+            "javascript" => Lang::JavaScript,
+            "css" => Lang::CSS,
+            _ => Lang::NoLanguage,
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct Props {
+    pub text: String,
+    pub title: String,
+    pub language: Option<String>,
+    pub stargazers_count: u32,
+    pub html_url: String
+}
+
+pub struct ProjectCard{
+    pub description: String,
+    pub language: Lang,
+    pub name: String,
+    pub stargazers_count: u32,
+    pub html_url: String
+}
+
+pub enum Msg {}
+
+impl Component for ProjectCard {
+    type Message = Msg;
+    type Properties = Props;
+
+    fn create(props: Self::Properties, _: ComponentLink<Self>) -> Self {
+        let mut lang: Lang = Lang::NoLanguage;
+
+        if let Some(language) = props.language {
+            lang = Lang::from(language);
+        }
+
+        Self {
+            description: props.text,
+            language: lang,
+            name: props.title,
+            stargazers_count: props.stargazers_count,
+            html_url: props.html_url
+        }
+    }
+
+    fn update(&mut self, _msg: Self::Message) -> ShouldRender {
+        true
+    }
+
+    fn change(&mut self, _: Self::Properties) -> ShouldRender {
+        true
+    }
+
+    fn rendered(&mut self, _first_render: bool) {
+
+    }
+
+    fn view(&self) -> Html {
+        let (_class_name, language_text) = self.language.markup_resources();
```

Comment:
If you are not using the first member of the tuple, you can just do:

```suggestion
        let (_, language_text) = self.language.markup_resources();
```

But as there's only two we are writing so much here, you can have it even more simple if you just do classic assign here:

```suggestion
        let language_text = self.language.markup_resources().1;
```